### PR TITLE
Fixed issue [security] #18514 : no CSRF control when delete token table

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: move to a post action forum, validate with getPost

### CODE REVIEW

**Comment by kevin-foster-uk** [comment]
I just merged your newer pull request: [Fixed issue [security] #18514 : no CSRF control when delete token table](https://github.com/LimeSurvey/LimeSurvey/pull/2760)

This pull request appears to be handling the same issue. Am I misunderstanding something? Why do we have multiple PR's for this?

**Comment by Shnoulle** [comment]
> I just merged your newer pull request: [Fixed issue [security] #18514 : no CSRF control when delete token table](https://github.com/LimeSurvey/LimeSurvey/pull/2760)
> 
> This pull request appears to be handling the same issue. Am I misunderstanding something? Why do we have multiple PR's for this?

master != 3.x-LTS

We don't merge fix from master to LTS or LTS to master.
Then : when fix is needed for 3LTS too : need to create 2 PR.

:)

**Comment by Shnoulle** [comment]
Still security issue in master since near 2 month now …

**Review by gabrieljenik — APPROVED** [review]
Code looks OK. Haven't tested it.

**Comment by olleharstedt** [comment]
@gabrieljenik Can your guys test this perhaps?

**Comment by Shnoulle** [comment]
PS : @kevin-foster-uk already merge it in 3LTS : https://github.com/LimeSurvey/LimeSurvey/pull/2760

**Comment by gabrieljenik** [comment]
> @gabrieljenik Can your guys test this perhaps?

Only kill function (delete participants table) seems to be impacted, right?

**Comment by Shnoulle** [comment]
> > @gabrieljenik Can your guys test this perhaps?
> 
> Only kill function (delete participants table) seems to be impacted, right?

I found only this one.

File: application/controllers/admin/Tokens.php [hunk 1]
```diff
@@ -2365,34 +2365,32 @@ public function kill(int $iSurveyId)
...
-            $archivedTokenSettings->tbl_name = $newtable;
-            $archivedTokenSettings->tbl_type = 'token';
-            $archivedTokenSettings->created = $DBDate;
-            $archivedTokenSettings->properties = $aData['thissurvey']['tokenencryptionoptions'];
-            $archivedTokenSettings->attributes = json_encode($aData['thissurvey']['attributedescriptions']);
-            $archivedTokenSettings->save();
-
-            //Remove any survey_links to the CPDB
-            SurveyLink::model()->deleteLinksBySurvey($iSurveyId);
+            return;
```
Comment (olleharstedt) [inline application/controllers/admin/Tokens.php:2372]: Hm shouldn't this be a no-permission error?

**Comment by olleharstedt** [comment]
Just one small question from me.

File: application/controllers/admin/Tokens.php (hunk 1 above)
Comment (Shnoulle) [reply application/controllers/admin/Tokens.php:2372]: No : before it's same link with or without &ok=1

Permission is check at https://github.com/LimeSurvey/LimeSurvey/blob/bc26b5b11aa0addae1ecf0cd453eca2ee0685eab/application/controllers/admin/Tokens.php#L1312

If Yii::app()->request->getPost('ok') : we do the actin, if !Yii::app()->request->getPost('ok') we show the confirm action form.

Like before except moving to POST action to have CRSF

Comment (olleharstedt) [reply application/controllers/admin/Tokens.php:2372]: K

**Pull request merged**